test/gtest/uct/test_sockaddr.cc

```diff
@@ -565,6 +564,8 @@ class test_uct_sockaddr : public uct_test {
         } else {
             self->del_user_data(sa_user_data);
         }
+
+        self->handle_client_connecting_status(status);
```

Comment:
why moving handle_client_connecting_status to the end fixed the issue? it just updates m_state

Reply:
there is a race condition between `m_state` and `m_ep_client_data` which can be observed in line 1094:
```
    wait_for_bits(&m_state, TEST_STATE_SERVER_REJECTED |
                            TEST_STATE_CLIENT_GOT_REJECT);
```

Reply:
So the race is because handle_client_connecting_status adds TEST_STATE_CLIENT_GOT_REJECT to m_state which is atomic
pls add comment before line 568, swap order of lines 562-563 and make counters like m_client_connect_cb_cnt also atomic:
```c
    volatile int           m_client_connect_cb_cnt,
                           m_server_connect_cb_cnt;
    volatile int           m_server_disconnect_cnt, m_client_disconnect_cnt;
```